action/protocol/staking/handler_candidate_transfer_ownership.go

```diff
@@ -0,0 +1,89 @@
+package staking
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/iotexproject/iotex-address/address"
+	"github.com/iotexproject/iotex-proto/golang/iotextypes"
+	"github.com/pkg/errors"
+
+	"github.com/iotexproject/iotex-core/action"
+	"github.com/iotexproject/iotex-core/action/protocol"
+	accountutil "github.com/iotexproject/iotex-core/action/protocol/account/util"
+)
+
+const (
+	handleCandidateTransferOwnership = "candidateTransferOwnership"
+)
+
+func (p *Protocol) handleCandidateTransferOwnership(ctx context.Context, act *action.CandidateTransferOwnership, csm CandidateStateManager,
+) (*receiptLog, []*action.TransactionLog, error) {
+	actCtx := protocol.MustGetActionCtx(ctx)
+	featureCtx := protocol.MustGetFeatureCtx(ctx)
+
+	log := newReceiptLog(p.addr.String(), handleCandidateTransferOwnership, featureCtx.NewStakingReceiptFormat)
+	_, fetchErr := fetchCaller(ctx, csm, big.NewInt(0))
+	if fetchErr != nil {
+		return log, nil, fetchErr
+	}
+
+	if err := p.validateCandidateTransferOwnership(ctx, act, csm, actCtx.Caller); err != nil {
+		return log, nil, err
+	}
+	candidate := csm.GetByOwner(actCtx.Caller).Clone()
+	if candidate.Identifier == nil || candidate.Identifier.String() == "" {
+		candidate.Identifier = candidate.Owner
+	}
+	candidate.Owner = act.NewOwner()
+	if err := csm.Upsert(candidate); err != nil {
+		return log, nil, csmErrorToHandleError(candidate.Owner.String(), err)
+	}
+	log.AddTopics(actCtx.Caller.Bytes(), act.NewOwner().Bytes())
+	return log, nil, nil
+}
+
+func (p *Protocol) validateCandidateTransferOwnership(_ context.Context, act *action.CandidateTransferOwnership,
+	csm CandidateStateManager, caller address.Address) ReceiptError {
+	//check if the candidate exists
+	candidate := csm.GetByOwner(caller)
+	if candidate == nil {
+		return &handleError{
+			err:           errors.New("candidate does not exist"),
+			failureStatus: iotextypes.ReceiptStatus_ErrCandidateNotExist,
+		}
+	}
+	//check if the new owner is self
+	if address.Equal(act.NewOwner(), caller) {
+		return &handleError{
+			err:           errors.New("new owner is the same as the current owner"),
+			failureStatus: iotextypes.ReceiptStatus_ErrUnauthorizedOperator,
+		}
+	}
+	//check the new owner is exist or is contract address
+	if acc, err := accountutil.LoadOrCreateAccount(csm.SM(), act.NewOwner()); err != nil || acc.IsContract() {
```

Comment:
I don't think it's necessary,  btw it also cannot reject future contract account

Reply:
after discuss offline, only transfer to an existed EOA